### PR TITLE
Parse command line and add helper methods for libclang interface

### PR DESCRIPTION
**Pull request by Burgos**
This adds the Isekai main program class which reads the command
line arguments and creates the CParser instance which will parse the
input C file and output the arithmetic circuits. Additionally, this PR introduces
helpers for abstracting some of the operations with libclang.

### CODE REVIEW

**Review by pcarinhas — APPROVED** [review]

File: src/isekai.cr [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        # Ignore overflow
+        property ignore_overflow = false
+    end
+
+    class ParserProgram
+        # Main 
+        def main
+            opts = ProgramOptions.new
+
+            OptionParser.parse! do |parser|
```
Comment (pcarinhas) [inline src/isekai.cr:39]: Please put some comments on what each option does.?

File: src/isekai.cr [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                parser.banner = "Usage: isekai [arguments] file"
+                parser.on("-c", "--cpparg=ARGS", "Extra arguments to clang")\
+                    { |args| opts.clang_args = args }
+                parser.on("-a", "--arith=FILE", "Arithmetic circuit output file")\
+                    { |file| opts.arith_file = file }
+                parser.on("-b", "--bool=FILE", "Boolean circuit output file")\
+                    { |file| opts.bool_file = file }
+                parser.on("-w", "--bit-width", "Bit width")\
```
Comment (Sp1r0u) [inline src/isekai.cr]: Can you explain what is the bit width option?

File: src/isekai.cr [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    class ParserProgram
+        # Main 
+        def main
+            opts = ProgramOptions.new
+
+            OptionParser.parse! do |parser|
+                parser.banner = "Usage: isekai [arguments] file"
+                parser.on("-c", "--cpparg=ARGS", "Extra arguments to clang")\
+                    { |args| opts.clang_args = args }
+                parser.on("-a", "--arith=FILE", "Arithmetic circuit output file")\
```
Comment (Sp1r0u) [inline src/isekai.cr:43]: How does the user define the format of the circuit (i.e. arithmetic or binary)?

File: src/isekai.cr (hunk 2 above)
Comment (Burgos) [reply src/isekai.cr]: I've added a bit more extended description in the `struct ProgramOptions` and I put a comment here to refer to that struct for more detailed descriptions. I also updated this brief description.

File: src/isekai.cr (hunk 3 above)
Comment (Burgos) [reply src/isekai.cr:43]: These are independent - if any `-a` or  `-b` set, the appropriate output will be created (which could be both).

**Comment by Burgos** [comment]
Updated with tests, makefile and readme file.

**Review by Sp1r0u — APPROVED** [review]

**Review by Sp1r0u — APPROVED** [review]

**Review by Sp1r0u — APPROVED** [review]

**Comment by Burgos** [comment]
Thanks! I'll merge after we merge the license change.

**Review by pcarinhas — APPROVED** [review]

File: src/isekai.cr (hunk 1 above)
Comment (Burgos) [reply src/isekai.cr:39]: I put the comment here to refer to `ProgramOptions` for more detailed description here, not to duplicate comments, is that OK?

**Pull request merged**